src/bin/tectonic.rs

```diff
@@ -237,8 +239,19 @@ fn main() {
     // something I'd be relatively OK with since it'd only affect the progam
     // UI, not the processing results).
 
-    let mut status =
-        TermcolorStatusBackend::new(ChatterLevel::from_str(&args.chatter_level).unwrap());
+    let chatter_level = ChatterLevel::from_str(&args.chatter_level).unwrap();
+    let use_cli_color = match &*args.cli_color {
+        "always" => true,
+        "auto" => atty::is(atty::Stream::Stdout),
+        "never" => false,
+        _ => unreachable!(),
+    };
```

Comment:
For what it's worth, I have a memory that the Rust CLI WG had some small recommended crate whose job was to help you decide whether to use color or not ... it also checked an environment variable of some kind, I think. But now I'm not able to dig it up, so whatever.